scrapi/harvesters/dataone.py

```diff
@@ -124,35 +126,32 @@ class DataOneHarvester(XMLHarvester):
             ('formatId', ("str[@name='formatId']/node()")),
             ('formatType', ("str[@name='formatType']/node()")),
             ('identifier', ("str[@name='identifier']/node()")),
-            ('investigator', "arr[@name='investigator']/str/node()"),
-            ('origin', "arr[@name='origin']/str/node()"),
-            ('isPublic', ("bool[@name='isPublic']/node()")),
             ('readPermission', "arr[@name='readPermission']/str/node()"),
             ('replicaMN', "arr[@name='replicaMN']/str/node()"),
             ('replicaVerifiedDate', "arr[@name='replicaVerifiedDate']/date/node()"),
             ('replicationAllowed', ("bool[@name='replicationAllowed']/node()")),
             ('numberReplicas', ("int[@name='numberReplicas']/node()")),
             ('preferredReplicationMN', "arr[@name='preferredReplicationMN']/str/node()"),
-            ('resourceMap', "arr[@name='resourceMap']/str/node()"),
             ('rightsHolder', ("str[@name='rightsHolder']/node()")),
             ('scientificName', "arr[@name='scientificName']/str/node()"),
             ('site', "arr[@name='site']/str/node()"),
             ('size', ("long[@name='size']/node()")),
-            ('sku', ("str[@name='sku']/node()")),
             ('isDocumentedBy', "arr[@name='isDocumentedBy']/str/node()"),
             ('serviceID', "str[@name='id']/node()")
         ),
         'freeToRead': {
             'startDate': ("bool[@name='isPublic']/node()", "date[@name='dateModified']/node()", lambda x, y: parse(y[0]).date().isoformat() if x else None)
         },
-        'contributors': ("str[@name='author']/node()", "str[@name='submitter']/node()", "arr[@name='origin']/str/node()", process_contributors),
+        'contributors': ("str[@name='author']/node()", "str[@name='submitter']/node()", "arr[@name='origin']/str/node()", "arr[@name='investigator']/str/node()", process_contributors),
         'uris': {
             'canonicalUri': ("str[@name='id']/node()", "//str[@name='dataUrl']/node()", lambda x, y: y[0] if 'http' in single_result(y) else x[0] if 'http' in single_result(x) else ''),
+            'objectUri': "str[@name='resourceMap']/node()".replace("doi:", "http://dx.doi.org/")
```

Comment:
it doesn't like  'objectUri': "str[@name='resourceMap']/node()", lambda x: x.replace('doi:', 'http://dx.doi.org/') because x is apparently a list object.